addons/monit/monit_build_configuration.pl

```diff
@@ -0,0 +1,130 @@
+my @configurations = split(/\,/, $configurations);
+
+foreach my $configuration ( @configurations ) {
+    die "Invalid configuration parameter '$configuration'\n" if !exists $CONFIGURATION_TO_TEMPLATE{$configuration};
+}
+
+
+generate_configurations();
+monit_syslog_configuration();
+print "\n\nAll set!\n\n";
+
+
+sub generate_configurations {
+    print "\n\nGenerating the following configuration files: \n";
+
+    foreach my $configuration ( @configurations ) {
+        my $template_file = catfile($MONIT_CHECKS_CONF_TEMPLATES_PATH,$CONFIGURATION_TO_TEMPLATE{$configuration} . $TEMPLATE_FILE_EXTENSION);
+        my $destination_file = catfile($MONIT_PATH,$CONFIGURATION_TO_TEMPLATE{$configuration} . $CONF_FILE_EXTENSION);
+        print " - $destination_file\n";
+
+        # Backing up existing configuration file (just in case)
+        move($destination_file, $destination_file . ".bak") if -e $destination_file;
+
+        # Handling pfdhcplisteners
+        my $pfdhcplisteners = handle_pfdhcplisteners();
+        # Handling domains (winbind configuration)
+        my $domains = handle_domains();
+
+        my $tt = Template->new(ABSOLUTE => 1);
+        my $freeradius_bin = ( $OS eq "rhel" ) ? "radiusd" : "freeradius";
+        my $vars = {
+            FREERADIUS_BIN      => $freeradius_bin,
+            EMAILS              => \@emails,
+            SUBJECT_IDENTIFIER  => $subject_identifier,
+            PFDHCPLISTENERS     => $pfdhcplisteners,
+            DOMAINS             => $domains,
+        };
+        $tt->process($template_file, $vars, $destination_file) or die $tt->error();
+    }
+}
+
+
+sub handle_pfdhcplisteners {
+    use pf::services::manager::pfdhcplistener;
+    my $self = pf::services::manager::pfdhcplistener->new(name => 'dummy');
+    my @managers = $self->pf::services::manager::pfdhcplistener::managers;
+    my @pfdhcplisteners = ();
+    push @pfdhcplisteners, $_->{name} foreach ( @managers );
+    return \@pfdhcplisteners;
+}
+
+sub handle_domains {
+    use pf::config;
+    use pf::services::manager::winbindd_child;
+    my %domains = ();
+    foreach my $domain ( keys(%pf::config::ConfigDomain) ) {
+        $domains{$domain} = pf::services::manager::winbindd_child->new(name => 'dummy', domain => "$domain")->pidFile;
+    }
+    return \%domains;
+}
+
+sub monit_syslog_configuration {
+    my $syslog_engine = ( $OS eq "rhel" ) ? "syslog" : "rsyslog";
+
+    # Generate syslog Monit configuration file
+    my $tt = Template->new(ABSOLUTE => 1);
+    my $vars = {
+        MONIT_LOG_FILE  => $MONIT_LOG_FILE,
+    };
+    $tt->process(catfile($MONIT_CONF_TEMPLATES_PATH, "syslog_monit" . $TEMPLATE_FILE_EXTENSION), $vars, "/etc/$syslog_engine.d/monit.conf") or die $tt->error();
+#    $tt->process($PF_PATH . "/addons/monit/" . "syslog_monit" . $TEMPLATE_FILE_EXTENSION, $vars, "/etc/$syslog_engine.d/monit.conf") or die $tt->error();
+
+    # Remove default Monit logging configuration file
+    unlink '/etc/monit.d/logging';
```

Comment:
Forget my comment... If we don't delete it then it will break stuff so we need to fix the path to ensure its deleted on debian also